frontend/reducers/sites.js

```diff
@@ -1,39 +1,59 @@
 import {
+  sitesFetchStartedType as SITES_FETCH_STARTED,
   sitesReceivedType as SITES_RECEIVED,
   siteAddedType as SITE_ADDED,
   siteUpdatedType as SITE_UPDATED,
   siteDeletedType as SITE_DELETED,
   siteBranchesReceivedType as SITE_BRANCHES_RECEIVED,
 } from '../actions/actionCreators/siteActions';
 
-const initialState = [];
+const initialState = { isLoading: false };
 
 export default function sites(state = initialState, action) {
   switch (action.type) {
 
+  case SITES_FETCH_STARTED:
+    return { isLoading: true }
+
   case SITES_RECEIVED:
-    return action.sites || initialState;
+    return { isLoading: false, data: action.sites || [] }
 
   case SITE_ADDED:
-    return action.site ? [...state, action.site] : state;
+    if (action.site) {
+      return {
+        isLoading: false,
+        data: state.data.concat(action.site),
+      }
+    } else {
+      return state
+    }
 
   case SITE_UPDATED:
-    return mapPropertyToMatchingSite(state, action.siteId, action.site);
+    return {
+      isLoading: false,
```

Comment:
Are there places where the site is updated, deleted, or the branches are received where the sites_fetch_started signal is called? I may be missing it. Otherwise, I'm not sure I understand why we're setting `isLoading` again here and the other places below.